code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm

```diff
@@ -261,3 +264,213 @@
+		/turf/open/floor/plating = 3, 
+		)
+	
+	weighted_feature_spawn_list = list(
+		/obj/structure/filingcabinet/chestdrawer = 1,
+		/obj/item/kirbyplants/random  = 2,
+		list(/obj/structure/rack , /obj/item/melee/spear/plugged_musket) = 1,
+		list(/obj/structure/rack , /obj/item/clothing/suit/armor/vest, /obj/item/clothing/head/helmet/riot) = 1,
+		/obj/structure/frame/machine = 1,
+		/obj/structure/frame/computer = 1,
+		/obj/effect/mine/stun	
+		)
+	
+	weighted_mob_spawn_list = list(
+		/mob/living/simple_animal/hostile/robot/burst = 2,
+		)
+/datum/dungeon_room_theme/maintenance/ancient_armory/pre_initialize()
+
+	. = ..()
+	for(var/i in 1 to 3)
+		if(prob(10))
+			weighted_feature_spawn_list[/obj/item/kitchen/knife/combat/bone]++
+		else
+			weighted_feature_spawn_list[/obj/item/melee/spear/bonespear/chitinspear]++
+
+///these don't actually spawn inside the suit storage and I think thats funny
+/datum/dungeon_room_theme/maintenance/hardsuit
+	weighted_feature_spawn_list = list(
+		list(/obj/machinery/suit_storage_unit,/obj/item/clothing/suit/space/hardsuit/mining),
+		list(/obj/machinery/suit_storage_unit,/obj/item/clothing/suit/space/hardsuit/ancient),
+		list(/obj/machinery/suit_storage_unit,/obj/item/clothing/suit/space/hardsuit/engine),
+		list(/obj/machinery/suit_storage_unit,/obj/item/clothing/suit/space/hardsuit/medical)
+		)
+
+	weighted_mob_spawn_list = list(
+		/mob/living/simple_animal/hostile/robot/burst = 2,
+		/mob/living/simple_animal/hostile/robot/burst = 2
+		)
+
+///this gun blows so much ass using it is actually more likely to get you killed
+/datum/dungeon_room_theme/maintenance/lasgun
+
+	weighted_mob_spawn_list = list(
+		/mob/living/simple_animal/hostile/robot/burst = 2,
+		/mob/living/simple_animal/hostile/robot/advanced/ranged = 1
+		)
+
+/datum/dungeon_room_theme/maintenance/lasgun/pre_initialize()
+	. = ..()
+	for(var/i in 1 to 5)
+		if(prob(10))
+			weighted_feature_spawn_list[/obj/item/gun/energy/laser/captain/scattershot ]++
+		else
+			weighted_feature_spawn_list[/obj/item/melee/spear/bonespear/chitinspear]++
+
+///lathes, nough said
+/datum/dungeon_room_theme/maintenance/autolathe
+	weighted_feature_spawn_list = list(
+		/obj/machinery/autolathe/hacked = 1,
+		/obj/machinery/autolathe = 1,
+		/obj/machinery/autolathe = 1,
+		/obj/item/stack/sheet/glass/fifty = 1,
+		/obj/item/stack/sheet/metal/fifty = 1,
+		/obj/item/stack/sheet/mineral/silver/fifty = 1
+		)
+///eskimo enemy variety, they came here for the winter
+/datum/dungeon_room_theme/maintenance/eskimo
+	weighted_mob_spawn_list = list(
+		/mob/living/simple_animal/hostile/skeleton/eskimo = 2,
+		/mob/living/simple_animal/hostile/skeleton/eskimo = 1
+		)
+///I really really really want to include a real chasm but god knows im not strong enough
+/datum/dungeon_room_theme/maintenance/chasm_fake
+	weighted_possible_floor_types = list(
+		/turf/open/floor/fakepit = 3,
+		/turf/open/floor/fakepit = 5,
+		/turf/open/floor/fakepit= 3, 
```

Comment:
what, why, why is this here 3 times with different weights

Reply:
im a silly little guy

Reply:
(I changed it)